app/siteHacks.js

```diff
@@ -8,78 +8,148 @@ const urlParse = require('./common/urlParse')
 const Filtering = require('./filtering')
 const {siteHacks} = require('../js/data/siteHacks')
 const appConfig = require('../js/constants/appConfig')
+const {makeJS} = require('./common/state/immutableUtil')
 
 const resourceName = 'siteHacks'
+let referralHeaders = null
 
-module.exports.init = () => {
-  if (!appConfig[resourceName].enabled) {
-    return
+const beforeHeaders = (details, isPrivate) => {
+  const mainFrameUrl = Filtering.getMainFrameUrl(details)
+  if (!mainFrameUrl) {
+    return {
+      resourceName
+    }
   }
 
-  Filtering.registerBeforeSendHeadersFilteringCB((details, isPrivate) => {
-    const mainFrameUrl = Filtering.getMainFrameUrl(details)
-    if (!mainFrameUrl) {
-      return {
-        resourceName
+  const domain = urlParse(mainFrameUrl).hostname.split('.').slice(-2).join('.')
```

Comment:
this doesn't work for domains that have subdomains, ex: `play.spotify.com`

Reply:
you can use `getBaseDomain` to get the full domain

Reply:
though it looks like it also does this on master

Reply:
yes this code is on the master as well, where I only moved this code into named funtion